modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ConstructMessageProcessor.java

```diff
@@ -0,0 +1,349 @@
+import org.mule.runtime.core.internal.policy.PolicyManager;
+import org.mule.runtime.extension.api.runtime.config.ConfigurationProvider;
+import org.mule.runtime.extension.api.runtime.operation.ExecutionContext;
+import org.mule.runtime.extension.api.runtime.operation.Interceptor;
+import org.mule.runtime.extension.api.runtime.operation.OperationExecutor;
+import org.mule.runtime.extension.api.runtime.operation.OperationExecutorFactory;
+import org.mule.runtime.module.extension.api.loader.java.property.OperationExecutorModelProperty;
+import org.mule.runtime.module.extension.api.runtime.privileged.ExecutionContextAdapter;
+import org.mule.runtime.module.extension.internal.runtime.DefaultExecutionContext;
+import org.mule.runtime.module.extension.internal.runtime.ExtensionComponent;
+import org.mule.runtime.module.extension.internal.runtime.LazyExecutionContext;
+import org.mule.runtime.module.extension.internal.runtime.execution.OperationArgumentResolverFactory;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ParameterValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ResolverSet;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import reactor.core.publisher.Mono;
+
+/**
+ * A {@link Processor} capable of executing extension operations.
+ * <p>
+ * It obtains a configuration instance, evaluate all the operation parameters and executes a {@link OperationModel} by using a
+ * {@link #operationExecutor}. This message processor is capable of serving the execution of any {@link OperationModel} of any
+ * {@link ExtensionModel}.
+ * <p>
+ * A {@link #operationExecutor} is obtained by testing the {@link OperationModel} for a {@link OperationExecutorModelProperty}
+ * through which a {@link OperationExecutorFactory} is obtained. Models with no such property cannot be used with this class. The
+ * obtained {@link OperationExecutor} serve all invocations of {@link #process(InternalEvent)} on {@code this} instance but will
+ * not be shared with other instances of {@link ConstructMessageProcessor}. All the {@link Lifecycle} events that {@code this}
+ * instance receives will be propagated to the {@link #operationExecutor}.
+ * <p>
+ * The {@link #operationExecutor} is executed directly but by the means of a {@link DefaultExecutionMediator}
+ * <p>
+ * Before executing the operation will use the {@link PolicyManager} to lookup for a {@link OperationPolicy} that must be applied
+ * to the operation. If there's a policy to be applied then it will interleave the operation execution with the policy logic
+ * allowing the policy to execute logic over the operation parameters, change those parameters and then execute logic with the
+ * operation response.
+ *
+ * @since 3.7.0
+ */
+public class ConstructMessageProcessor extends ExtensionComponent<ConstructModel>
```

Comment:
Is this OperationMessageProcessor renamed? Operations are not constructs

Reply:
not exactly the same, lots of similar things though. I can create a ticket of technical debt generify even more the ExtensionComponent and make Construct and Operation MP thiner.

Reply:
that's a dangerous tech debt to live with. Reactive engine is still an active area of development. People still working on it are not going to realize they have to change things in both places